dockstore-webservice/src/main/java/io/dockstore/webservice/core/Entry.java

```diff
@@ -85,11 +85,13 @@
         @NamedQuery(name = "Entry.getGenericEntryByAlias", query = "SELECT e from Entry e JOIN e.aliases a WHERE KEY(a) IN :alias"),
         @NamedQuery(name = "io.dockstore.webservice.core.Entry.findCollectionsByEntryId", query = "select new io.dockstore.webservice.core.CollectionOrganization(col.id, col.name, col.displayName, organization.id, organization.name, organization.displayName) from Collection col join col.entries as entry join col.organization as organization where entry.entry.id = :entryId and organization.status = 'APPROVED'"),
         @NamedQuery(name = "Entry.getCollectionWorkflows", query = "SELECT new io.dockstore.webservice.core.CollectionEntry(w.id, w.dbUpdateDate, 'workflow', w.sourceControl, w.organization, w.repository, w.workflowName) from BioWorkflow w, Collection col join col.entries as e where col.id = :collectionId and e.version is null and w.id = e.entry.id and w.isPublished = true"),
+        @NamedQuery(name = "Entry.getWorkflowsLength", query = "SELECT COUNT(w.id) FROM BioWorkflow w, Collection col join col.entries as e where col.id = :collectionId and e.version is null and w.id = e.entry.id and w.isPublished = true"),
```

Comment:
Regarding `e.version is null`, I saw your response to my comment in the other PR flash by, but forget the details.

I'm still not convinced this is right. If you create a collection where you add one version, not a workflow, and add nothing else, to it, what does the length come back as? 1 or 0?

Reply:
Is adding a version without an entry even possible?

Reply:
> Is adding a version without an entry even possible?

I thought you added the feature to pin a specific version of a workflow to a collection. :) 

My concern is that if you do that, then `e.version` will not be null, and it won't show up in the count.

Reply:
(Also concerned about this but don't have anything to add)

Reply:
If memory serves. u add either just the entry or entry + version.